Source/Core/Core/PowerPC/PowerPC.h

```diff
@@ -303,7 +303,7 @@ inline UReg_XER GetXER()
   xer |= PowerPC::ppcState.xer_stringctrl;
   xer |= PowerPC::ppcState.xer_ca << XER_CA_SHIFT;
   xer |= PowerPC::ppcState.xer_so_ov << XER_OV_SHIFT;
-  return xer;
+  return UReg_XER{xer};
```

Comment:
I'm curious, do compilers let you omit the type name here (and simply write `return {xer};`)?
Not implying you _should_ write it like that, though.

Reply:
They don't if the constructor being invoked is explicit. Part of the restrictions `explicit` imposes is that implicit copy construction is also prevented, and `return <expression>;` would perform copy initialization here (since we return by value). Given `return {contents};` would be an initializer list and not the actual type, it would also be flagged by the compiler.

Reply:
With `SetXER({xer})` it _should_ become uniform initialization with type deduction, right? Was wondering if the same could happen here; but I guess that answers it. Thanks.